nebula_common/include/nebula_common/util/expected.hpp

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <variant>
+
+namespace nebula
+{
+namespace util
+{
+
+/// @brief A poor man's backport of C++23's std::expected.
+///
+/// At any given time, holds exactly one of: expected value, or error value.
+/// Provides functions for easy exception handling.
+///
+/// More info here: https://en.cppreference.com/w/cpp/utility/expected
+///
+/// @tparam T Type of the expected value
+/// @tparam E Error type
+template <typename T, typename E>
+struct expected
+{
+  bool has_value() { return std::holds_alternative<T>(value_); }
+
+  T value() { return std::get<T>(value_); }
```

Comment:
What if no value is present?

Reply:
Addressed in [14ab32d](https://github.com/tier4/nebula/pull/131/commits/14ab32dc57a5d6877223b065138295531affe5c4) by defining a `bad_expected_access` error similar to the one in C++23 and adding doc comments for explanation.